src/v2/Apps/Order/Components/__tests__/AddressModal.jest.tsx

```diff
@@ -59,11 +59,28 @@ describe("AddressModal", () => {
     expect(wrapper.text()).toContain("Edit address")
     expect(wrapper.find("input").length).toBe(7)
     expect(wrapper.find("select").length).toBe(1)
+    expect(wrapper.find("Checkbox[data-test='setAsDefault']").length).toBe(1)
     expect(wrapper.find("Text[data-test='deleteButton']").length).toBe(1)
     expect(wrapper.find("Button[data-test='saveButton']").length).toBe(1)
   })
 
-  it("renders AddModal with the title, input fields and button", () => {
+  it("renders EditModal withoun checkbox when address is default", () => {
+    const wrapper = getWrapper({
+      ...testAddressModalProps,
+      address: {
+        ...savedAddress,
+        isDefault: true,
+      },
+    })
+    expect(wrapper.text()).toContain("Edit address")
+    expect(wrapper.find("input").length).toBe(7)
+    expect(wrapper.find("select").length).toBe(1)
+    expect(wrapper.find("Checkbox[data-test='setAsDefault']").length).toBe(0)
+    expect(wrapper.find("Text[data-test='deleteButton']").length).toBe(1)
+    expect(wrapper.find("Button[data-test='saveButton']").length).toBe(1)
```

Comment:
```suggestion
    expect(wrapper.find("Checkbox[data-test='setAsDefault']").length).toBe(0)
```

Not blocking but fairly important: The majority of these assertions are already covered in other tests. The only one I see that is important here is to verify the checkbox isn't appearing in these conditions. 

Having more assertions here than necessary will lead to the test requiring changes for unrelated reasons.